org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/linkedediting/LinkedEditingTest.java

```diff
@@ -102,44 +96,8 @@ protected boolean condition() {
 			annotations.put(model.getPosition(annotation), annotation);
 		}
 
-		Annotation annotation = annotations.get(new org.eclipse.jface.text.Position(10, 4));
+		Annotation annotation = annotations.get(new LinkedPosition(sourceViewer.getDocument(), 10, 4, 0));
 		Assert.assertNotNull(annotation);
-		assertEquals(LSPLinkedEditingReconcilingStrategy.LINKEDEDITING_ANNOTATION_TYPE, annotation.getType());
-
-		Assert.assertTrue(viewer.isEditable());
-
-		try {
-			// Test initial insert
-			CompletableFuture<DidChangeTextDocumentParams> didChangeExpectation = new CompletableFuture<DidChangeTextDocumentParams>();
-			MockLanguageServer.INSTANCE.setDidChangeCallback(didChangeExpectation);
-			String text = "w";
-			String expectedChangeText = viewer.getDocument().get().replace("body>", "wbody>");
-
-			viewer.getTextWidget().replaceTextRange(10, 0, text);
-			
-			DidChangeTextDocumentParams lastChange = didChangeExpectation.get(1000, TimeUnit.MILLISECONDS);
-			assertNotNull(lastChange.getContentChanges());
-			assertEquals(1, lastChange.getContentChanges().size());
-			TextDocumentContentChangeEvent change0 = lastChange.getContentChanges().get(0);
-			assertEquals(expectedChangeText, change0.getText());
-			assertEquals(expectedChangeText, viewer.getDocument().get());
-			
-			// Test additional insert
-			didChangeExpectation = new CompletableFuture<DidChangeTextDocumentParams>();
-			MockLanguageServer.INSTANCE.setDidChangeCallback(didChangeExpectation);
-			expectedChangeText = viewer.getDocument().get().replace("body>", "wbody>");
-
-			viewer.getTextWidget().replaceTextRange(11, 0, text);
-			
-			lastChange = didChangeExpectation.get(1000, TimeUnit.MILLISECONDS);
-			assertNotNull(lastChange.getContentChanges());
-			assertEquals(1, lastChange.getContentChanges().size());
-			change0 = lastChange.getContentChanges().get(0);
-			assertEquals(expectedChangeText, change0.getText());
-			assertEquals(expectedChangeText, viewer.getDocument().get());
-		} catch (Exception e) {
-			e.printStackTrace();
-			fail(e.getMessage());
-		}
```

Comment:
I removed that because we basically only need to verify that link edit mode is on, we then fully rely on it.
The LinkedModeUI is actually triggering an extra change instead of using a documentCommand as done previously. We may want to improve it (in JFace) at some point).